tiny_cnn/layers/dropout_layer.h

```diff
@@ -134,22 +142,26 @@ class dropout_layer : public layer {
 
     std::string layer_type() const override { return "dropout"; }
 
-    const std::vector<uint8_t>& get_mask(cnn_size_t worker_index) const {
-        return dropout_layer_worker_storage_[worker_index].mask_;
+    const std::vector<uint8_t>& get_mask(cnn_size_t worker_index, cnn_size_t sample_index) const {
+        return dropout_layer_worker_storage_[worker_index].mask_[sample_index];
     }
 
     virtual void set_worker_count(cnn_size_t worker_count) override {
         Base::set_worker_count(worker_count);
         dropout_layer_worker_storage_.resize(worker_count);
 
         for (dropout_layer_worker_specific_storage& dws : dropout_layer_worker_storage_) {
-            dws.mask_.resize(in_size_);
+            if (dws.mask_.empty()) {
+                dws.mask_.resize(1, std::vector<uint8_t>(in_size_));
```

Comment:
@reunanen
Ah I misunderstood, you are correct. it doesn't require a lock while resizing.